nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/SetSystemPropertiesAction.java

```diff
@@ -52,14 +66,53 @@ public Topology getTopology(Map<String, String> options) {
 
     public static class SetSystemPropertyComputation extends AbstractBulkComputation {
 
+        public static final String NOTIFY = "param-notify";
+
         public SetSystemPropertyComputation() {
             super(ACTION_NAME);
         }
 
         @Override
         protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
-            ids.forEach(id -> properties.forEach((k, v) -> session.setDocumentSystemProp(new IdRef(id), k, v)));
-            session.save();
+            Map<String, Serializable> actualProperties = new HashMap<>(properties);
+            String eventId = (String) actualProperties.remove(NOTIFY);
+            Collection<DocumentRef> refs = ids.stream().map(IdRef::new).collect(Collectors.toList());
+            for (Iterator<DocumentRef> it = refs.iterator(); it.hasNext();) {
+                DocumentRef ref = it.next();
+                for (Entry<String, Serializable> entry : actualProperties.entrySet()) {
+                    try {
+                        session.setDocumentSystemProp(ref, entry.getKey(), entry.getValue());
+                    } catch (Exception e) {
+                        it.remove();
+                        // TODO send to error stream
+                        getLog().warn("Cannot set system property: " + entry.getKey() + " on: " + ref.toString(), e);
+                    }
+                }
+            }
+            try {
+                session.save();
+                if (eventId != null && !refs.isEmpty()) {
+                    fireEvent(session, eventId, refs);
+                }
+            } catch (PropertyException e) {
+                // TODO send to error stream
+                getLog().warn("Cannot save session", e);
```

Comment:
`getLog` -> direct static logger

Reply:
there is a ticket open for this, since it's not merged yet, I supposed it would be dealt in the corresponding PR